Finished initial refactor.

tritoke opened the pull request:
Also fixed a bug where $? wasn't expanded properly.
And generally attempted to make the code more idiomatic.

Commit:
Finished initial refactor.

Also fixed a bug where $? wasn't expanded properly.
And generally attempted to make the code more idiomatic.

chorman0773 commented on .gitignore:
```diff
@@ -4,7 +4,10 @@
 # Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
 # More information here https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html
-# Cargo.lock
+Cargo.lock
```
Cargo.lock should be checked out for executables.

tritoke replied:
sorry I'm not sure I understand, should it be included or not?

chorman0773 replied:
It should not be in `.gitignore`.

chorman0773 replied:
```suggestion
```

tritoke replied:
Fixed in commit #3cdd8e5

chorman0773 reviewed (changes requested):
Cargo.lock should be checked out for executables. Otherwise, looks good

Commit:
Removed Cargo.lock from .gitignore.

Fix from review:
https://github.com/LightningCreations/lc-make/pull/3#pullrequestreview-682325608
